rust/origen_metal/src/stil/stil.pest

```diff
@@ -274,8 +274,8 @@ scan_inversion_val = { ("0" | "1") }
 pattern_block = { "Pattern" ~ name ~ "{" ~ (label? ~ time_unit)? ~ pattern_statement* ~ "}" }
 label = { (string | name) ~ ":" }
 time_unit = { "TimeUnit" ~ time_expr ~ EOS }
-pattern_statement = { label | vector_with_comment | vector | waveform_statement | condition | call | macro_statement | loop_statement |
-                      match_loop | goto | breakpoint | iddq | stop_statement | scan_chain_statement | annotation }
+pattern_statement = !{ label | vector_with_comment | vector | waveform_statement | condition | call | macro_statement |loop_statement_with_comment | 
```

Comment:
What's the `!` added here for, is this fixing a bug?

Reply:
To get the loop_with_comment, I had to make it atomic (@), the ! (non-atomic) prevents the cascading effect into the pattern_statement that's included in the loop_with_comment rule

Reply:
Ah, I see, didn't realize `!` had this additional meaning, I was reading it as pattern_statement = not any of these!